src/model-typescript.ts

```diff
@@ -257,4 +238,211 @@ export class ModelTypeScript {
+   *   DataType, a string or a type-description object.
+   * @param options These options are merged with the default define options provided to the Sequelize constructor
+   */
+  static init<MS extends ModelStatic, M extends InstanceType<MS>>(
+    this: MS,
+    attributes: ModelAttributes<
+      M,
+      // 'foreign keys' are optional in Model.init as they are added by association declaration methods
+      Optional<Attributes<M>, BrandedKeysOf<Attributes<M>, typeof ForeignKeyBrand>>
+    >,
+    options: InitOptions<M>,
+  ): MS {
+    if (isDecoratedModel(this)) {
+      throw new Error(`Model.init cannot be used if the model uses one of Sequelize's decorators. You must pass your model to the Sequelize constructor using the "models" option instead.`);
+    }
+
+    if (!options.sequelize) {
+      throw new Error('Model.init expects a Sequelize instance to be passed through the option bag, which is the second parameter.');
+    }
+
+    initModel(this, attributes, options);
+
+    return this;
+  }
+
+  static getIndexes(): readonly IndexOptions[] {
+    return this.modelDefinition.getIndexes();
+  }
+
+  /**
+   * Unique indexes that can be declared as part of a CREATE TABLE query.
+   *
+   * @deprecated prefer using {@link getIndexes}, this will eventually be removed.
+   */
+  static get uniqueKeys() {
+    const indexes = this.getIndexes();
+    const uniqueKeys = Object.create(null);
+
+    // TODO: "column" should be removed from index definitions
+    const supportedOptions = ['unique', 'fields', 'column', 'name'];
+
+    for (const index of indexes) {
+      if (!index.unique) {
+        continue;
+      }
+
+      if (!index.name) {
+        continue;
+      }
+
+      if (!index.fields) {
+        continue;
+      }
+
+      if (!index.fields.every(field => typeof field === 'string')) {
+        continue;
+      }
+
+      if (!Object.keys(index).every(optionName => supportedOptions.includes(optionName))) {
+        continue;
+      }
+
+      uniqueKeys[index.name] = index;
+    }
+
+    return uniqueKeys;
+  }
+
+  // TODO [>7]: Remove this
+  private static get _indexes(): never {
+    throw new Error('Model._indexes has been replaced with Model.getIndexes()');
+  }
+
+  /**
+   * Refreshes the Model's attribute definition.
+   *
+   * @deprecated use {@link modelDefinition}.
+   */
+  static refreshAttributes(): void {
+    this.modelDefinition.refreshAttributes();
+  }
+
+  static assertIsInitialized(): void {
+    if (!this.isInitialized()) {
+      throw new Error(`Model "${this.name}" has not been initialized yet. You can check whether a model has been initialized by calling its isInitialized method.`);
+    }
+  }
+
+  static isInitialized(): boolean {
+    // @ts-expect-error -- getModelDefinition expects ModelStatic
+    return hasModelDefinition(this);
+  }
+
+  /**
+   * Get the table name of the model, taking schema into account. The method will return The name as a string if the model
```

Comment:
```suggestion
   * Get the table name of the model, taking schema into account. The method will return the name as a string if the model
```

Reply:
The documentation was completely outdated, I've updated it